### DISCUSSION TITLE
problemi di avvio di zigbee2mqtt

### DISCUSSION THREAD

**Discussion by reclus23** [indomus/forum#2995] [HUB: Home Assistant]
ciao,
ho appena acquistato la skyconnect e l'ho connessa ad HA che gira su macchina virtuale su nas.
Prima di connetterla, ho aggiornato il firmware della chiavetta e tramite il sito di skyconnect ho messo automaticamente l'ultimo.

nel nas, poi,  - macchina virtuale - ho montato la chiavetta).

In HA-sistema-hardware ho copiato la destinazione della chiavetta ed ho fatto la seguente compilazione yaml:

```data_path: /config/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt//:192.168.x.xx:1883
  user: mqtt_user
  password: xxxxxxxxxxxx
serial:
  port: >-
    /dev/serial/by-id/usb-Nabu_Casa_SkyConnect_v1.0_c861f9f67cd8ed118ecd746162c613ac-if00-port0
  adapter: ezsp
advanced:
  log_level: warn
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental: {}
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false```

(spero di aver fatto bene il markdown)

l'add-on si avvia (pallino verde) ma ricevo questo errore:

Zigbee2MQTT:error 2024-03-24 10:52:10: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-03-24 10:52:10: Failed to start zigbee
Zigbee2MQTT:error 2024-03-24 10:52:10: Exiting...
Zigbee2MQTT:error 2024-03-24 10:52:10: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.18.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.18.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.


Per favore, potreste indicarmi l'errore?
grazie

**Comment by reclus23** [comment]
non ho fatto bene il markdown..non so come, ci è finito dentro anche la seconda parte della mia richiesta..scusate

**Reply by snakuzzo** [reply]
A parte il markdown sbagliato... c'è poco da dire.
Non vede la chiavetta.

La skyconnect monta il chip EFR32MG21, quindi devi passargli 

 Devi assicurarti che sulla VM di HA OS sia correttamente montate e devi controllare bene il path, perché magari è diverso da quello che hai scritto.

**Comment by reclus23** [comment]
ciao snakuzzo grazie per l'aiuto. nella VM di HA alla voce usb mi trova soltanto silicon labs cp210x usb to uart bridge.

cosa intendevi quando hai scritto "quindi devi passargli"?  devo installare sul nas i drivers? (ma se in VM di Has alla voce hardware trovo la usb, non vuol dire che è stata correttamente riconosciuta e perfettamente funzionante?)
Il path che metto in zigbee2mqtt è quello che copio direttamente da impostazioni - hardware..
grazie

**Reply by snakuzzo** [reply]
Non ho parlato né di driver né di altro.
Per "passargli" intendevo dire "indicare in configurazione".
Comunque se sei sicuro del path, non vedo nulla che non vada a livello di configurazione.
Domanda...sei sicuro di non aver attivato ZHA?
Se così fosse, dovresti disabilitarlo.

**Comment by reclus23** [comment]
ZHA è disabilitato.
Per vedere se il path sia corretto è sufficiente che copio tutta la stringa che mi viene data in HA-Impostazioni-Hardware, giusto?

Può c'entrare qualcosa il fatto che il nas abbia soltanto una usb e dovendo avere anche il cavo dell'ups, ho messo nella usb del nas una multi usb? (però il nas mi indica correttamente quale usb voglia montare, se ups o silicon labs...)

**Reply by snakuzzo** [reply]
Solitamente la chiavetta viene montata sotto `/dev/ttyACM0`

![image](https://github.com/indomus/forum/assets/7205371/c0f02484-3d41-40fb-878e-1a40d1015057)

**Reply by reclus23** [reply]
io ho:
Percorso del dispositivo: /dev/ttyUSB0
ID:
/dev/serial/by-id/usb-Nabu_Casa_SkyConnect_v1.0_c861f9f67cd8ed118ecd746162c613ac-if00-port0

comunque ho provato anche con /dev/ttyACM0 e ricevo sempre lo stesso errore..

**Reply by reclus23** [reply]
facendo la ricerca ho trovato un paio di video che dicono che per il synology bisogna installare i drivers...
farò anche quel tentativo..

**Comment by reclus23** [comment]
ciao aggiorno il topic per dire che sono riuscito ad aggiungere i drivers in synology e la chiavetta ora è correttamente vista. da terminale appare e lanciando il comando ls viene ttyUSB0.
lanciando il comando ls -lart /dev/ttyUSB0 viene fuori 1 root radio 188

Come ulteriore conferma, ho installato ZHA e funziona. (dopo che ho visto che funzionava l'ho disinstallato)

Zigbee2mqtt, invece, mi riporta sempre lo stesso errore. A questo punto è questione di configurazione. 
Ho disinstallato zigbee2mqtt, ho cancellato tramite file editor tutte le cartelle riconducibili a zigbee2mqtt e l'ho reinstallato.
Niente, mi ritorna sempre lo stesso errore e non capisco dove sbaglio..

**Reply by reclus23** [reply]
chiudo perchè ho trovato il problema. stavo utilizzando non l'ultima versione di zigbee2mqtt...con l'ultima, nessun problema.
